src/Jellyfin.Database/Jellyfin.Database.Providers.Sqlite/PragmaConnectionInterceptor.cs

```diff
@@ -0,0 +1,80 @@
+using System.Data.Common;
+using System.Globalization;
+using System.Text;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+
+namespace Jellyfin.Database.Providers.Sqlite;
+
+/// <summary>
+/// Injects a series of PRAGMA on each connection starts.
+/// </summary>
+public class PragmaConnectionInterceptor : DbConnectionInterceptor
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PragmaConnectionInterceptor"/> class.
+    /// </summary>
+    /// <param name="cacheSize">Cache size.</param>
+    /// <param name="lockingMode">Locking mode.</param>
+    /// <param name="journalSizeLimit">Journal Size.</param>
+    /// <param name="pageSize">Page Size.</param>
+    public PragmaConnectionInterceptor(int? cacheSize, string lockingMode, int? journalSizeLimit, int? pageSize)
+    {
+        CacheSize = cacheSize;
+        LockingMode = lockingMode;
+        JournalSizeLimit = journalSizeLimit;
+        PageSize = pageSize;
+    }
+
+    private int? CacheSize { get; }
+
+    private string LockingMode { get; }
+
+    private int? JournalSizeLimit { get; }
+
+    private int? PageSize { get; }
+
+    private string? InitialCommand { get; set; }
+
+    /// <inheritdoc/>
+    public override void ConnectionOpened(DbConnection connection, ConnectionEndEventData eventData)
```

Comment:
You also need to override ConnectionOpenedAsync

Reply:
```
    /// <inheritdoc/>
    public override Task ConnectionOpenedAsync(
        DbConnection connection,
        ConnectionEndEventData eventData,
        CancellationToken cancellationToken = default)
    {
        var baseConnection = base.ConnectionOpenedAsync(connection, eventData, cancellationToken);

        using (var command = connection.CreateCommand())
        {
#pragma warning disable CA2100 // Review SQL queries for security vulnerabilities
            command.CommandText = InitialCommand ??= BuildCommandText();
#pragma warning restore CA2100 // Review SQL queries for security vulnerabilities
            command.ExecuteNonQueryAsync(cancellationToken);
        }

        return baseConnection;
    }
```
Should do it